### PR TITLE
[RecoPixelVertexing/PixelVertexFinding] updated BuildFile.xml and added plugins/

### PR DESCRIPTION
**Pull request by missirol**
#### PR description:

This PR suggests a small restructuring of the package `RecoPixelVertexing/PixelVertexFinding`.
I was trying to use the `PVClusterComparer` class in a separate CMSSW (user-)package,
but realized I couldn't do it without modifying the file `RecoPixelVertexing/PixelVertexFinding/BuildFile.xml` (adding the `export` directive).
While looking into this, I noticed `src/` contained both standalone classes and plugins, so I moved the latter ones under `plugins/` (and removed the file `src/SealModules.cc`),
as suggested in the [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html).

These changes are purely technical; no differences in the comparisons are expected.

#### PR validation:

Successfully ran a few standard workflows (for example, `runTheMatrix.py -l 11634.0`).

#### if this PR is a backport please specify the original PR:

Not a backport, and no backport of this is planned.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: RecoPixelVertexing/PixelVertexFinding/plugins/PixelVertexProducer.cc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "RecoPixelVertexing/PixelVertexFinding/interface/PixelVertexProducer.h"
+#include "RecoPixelVertexing/PixelVertexFinding/plugins/PixelVertexProducer.h"
```
Comment (slava77) [inline RecoPixelVertexing/PixelVertexFinding/plugins/PixelVertexProducer.cc]: please take this update as an opportunity to merge PixelVertexProducer.h contents into .cc file, which is a recommended approach

Comment (missirol) [reply RecoPixelVertexing/PixelVertexFinding/plugins/PixelVertexProducer.cc]: implemented this in 356519de19f

**Comment by perrotta** [comment]
please test

**Comment by fabiocos** [comment]
@missirol this is a useful move, thanks, along the lines discussed with @smuzaffar

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass and show no differences

**Comment by fabiocos** [comment]
+1

**Comment by missirol** [comment]
@perrotta @slava77 @fabiocos 
contrary to what I originally wrote in the description of this PR, I realized I could use a backport of this to 11_0_X; would that be allowed?
(if so, I will submit the backport PR, and add an edit to the description of this one)

**Comment by slava77** [comment]
the PR looks technical, why is the backport needed? Is there a use case from it for production?

**Comment by missirol** [comment]
@slava77 no production use case, it was just to simplify a user's recipe for HLT Phase-2 studies that are currently being done in 11_0_X (I thought a backport of this technical change was harmless, but it does not really matter). Thanks!

**Comment by fabiocos** [comment]
@missirol a backport looks harmless, but we usually avoid useless ones, unless there is a real serious use case. Is this the case here?

**Comment by missirol** [comment]
@fabiocos no problem. as i tried to write, it is not necessary. thanks!

**Pull request merged**